src/shared/switch-root.c

```diff
@@ -42,11 +41,17 @@ int switch_root(const char *new_root,
                 return 0;
 
         /* Check if we shall remove the contents of the old root */
-        old_root_remove = in_initrd();
-        if (old_root_remove) {
-                old_root_fd = open("/", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_NOCTTY|O_DIRECTORY);
-                if (old_root_fd < 0)
-                        return log_error_errno(errno, "Failed to open root directory: %m");
+        old_root_fd = open("/", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_NOCTTY|O_DIRECTORY);
+        if (old_root_fd < 0)
+                log_warning_errno(errno, "Failed to open root directory: %m");
```

Comment:
hmm, this looks like something we shouldn't tape over, but just fail. not being able to open the root dir, is just very strange...

(also, if we ignore it, we should say, `", ignoring"` in the log message. But I think we should not ignore it)

(btw, O_NONBLOCK should go, that makes no sense with O_DIRECTORY)

(and O_NOCTTY similar)

Reply:
I made it non-fatal because it's just for cleanup. It doesn't really matter, since this is unlikely to fail. I'll update it be be fatal.